### PR TITLE
Handle tombstone in combined iterator and add tests

### PR DESCRIPTION
**Pull request by guy-har**

### CODE REVIEW

File: graveler/combined_iterator.go [hunk 1]
```diff
@@ -46,7 +53,6 @@ func (c *combinedIterator) Next() bool {
 		// value of iterA < value of iterB
 		c.iterB.Next()
```
Comment (nopcoder) [inline graveler/combined_iterator.go:54]: use nextA/nextB for all Next() calls

File: graveler/combined_iterator.go [hunk 2]
```diff
@@ -18,23 +22,26 @@ func NewCombinedIterator(iterA, iterB ValueIterator) ValueIterator {
 	}
 }
 
-func (c *combinedIterator) Next() bool {
-	// call next with the relevant iterators
+// advanceInnerIterator advances the inner iterators and returns true if has more
+func (c *combinedIterator) advanceInnerIterator() bool {
```
Comment (nopcoder) [inline graveler/combined_iterator.go:26]: optional: advanceInnerIterator -> advanceInnerIterators

File: graveler/combined_iterator.go (hunk 1 above)
Comment (guy-har) [reply graveler/combined_iterator.go]: changed, thanks

File: graveler/combined_iterator.go (hunk 2 above)
Comment (guy-har) [reply graveler/combined_iterator.go]: 👍

**Review by nopcoder — APPROVED** [review]

**Pull request merged**